dll/win32/shell32/shldataobject.cpp

```diff
@@ -36,7 +36,7 @@ HRESULT WINAPI SHGetAttributesFromDataObject(IDataObject* pDataObject, DWORD dwA
 {
     DWORD dwAttributes = 0;
     DWORD cItems = 0;
-    HRESULT hr = S_OK;
+    HRESULT hr = S_OK, hr2;
```

Comment:
Since we're C++, I would suggest you just declare the hr2 at the point of usage (especially since you're using hr2 as a debug diagnostic tool):
```suggestion
    HRESULT hr = S_OK;
```